### DISCUSSION TITLE
RTK-Q multiple requests with queries already created

### DISCUSSION THREAD

**Discussion by dinoiam** [reduxjs/redux-toolkit#2046] [Q&A]
Hi, I was wondering if there is a way to perform multiple queries reusing queries already created. For example here instead of writing ```await fetchWithBQ(`user/${user.id}/posts`)``` could I do something like 
```await _queryApi.dispatch(fetchUserPosts.initiate({ userId }))``` ? The last doubt is about  `providesTags`, does it works in this case? Thanks in advance.

``` 
const api = createApi({
  baseQuery: fetchBaseQuery({ baseUrl: '/ ' }),
  endpoints: (build) => ({
    getRandomUserPosts: build.query<Post, void>({
      async queryFn(_arg, _queryApi, _extraOptions, fetchWithBQ) {
        // get a random user
        const randomResult = await fetchWithBQ('users/random')
        if (randomResult.error) throw randomResult.error
        const user = randomResult.data as User
        const result = await fetchWithBQ(`user/${user.id}/posts`)
        return result.data
          ? { data: result.data as Post }
          : { error: result.error as FetchBaseQueryError }
      },
    }),
  }),
})
```

**Comment by panoramix360** [comment]
Hi @dinoiam, did you get any answer or how to solve it?

thank you!

**Reply by dinoiam** [reply]
@panoramix360 no answers yet sorry. I'm doing like the example above.
`const result = await fetchWithBQ("user/${user.id}/posts/")`

**Reply by panoramix360** [reply]
Hi @dinoiam, I managed to do what you wanted, but it's not recommended because of cache duplication.

I asked this question yesterday, you can read more details [here](https://github.com/reduxjs/redux-toolkit/discussions/2972)

I hope that helps! 😃
